### PR TITLE
Unify service endpoint formatting through the project

### PR DESCRIPTION
**Pull request by bryk**
Now raw data is gathered on the backend and rendered on the frontend.

This PR looks large, but is is mostly moving things here and there.

### CODE REVIEW

**Comment by bryk** [comment]
@zreigz Can you review?

**Comment by zreigz** [comment]
Yes, last review in this year :)

File: src/app/backend/replicasetdetail.go [hunk 1]
```diff
@@ -211,3 +229,41 @@ func getRestartCount(pod api.Pod) int {
...
 	return restartCount
 }
+
+// Returns internal endpoint name for the given service properties, e.g.,
+// "my-service.namespace 80/TCP" or "my-service 53/TCP,53/UDP".
+func getInternalEndpoint(serviceName string, namespace string, ports []api.ServicePort) Endpoint {
+
+	name := serviceName
+	if namespace != api.NamespaceDefault {
+		name = name + "." + namespace
```
Comment (zreigz) [inline src/app/backend/replicasetdetail.go]: Very good practice is not using ```+``` for string concatenation.
Strings are immutable in Go - each time a string is assigned to a variable a new address is assigned in memory to represent the new value.  As the strings you are manipulating become large this process becomes increasingly slow. In fact, this method of naive appending is O(N2). (The same is in Java so StringBuffer is used instead ```+=```)
Here you should also use Buffer:
```
	bufferName := bytes.NewBufferString(name)
	bufferName.WriteString(".")
    bufferName.WriteString(namespace)
	name = bufferName.String()
```

**Comment by zreigz** [comment]
Rest looks good :)

**Comment by zreigz** [comment]
PTAL

File: src/app/backend/replicasetdetail.go (hunk 1 above)
Comment (bryk) [inline src/app/backend/replicasetdetail.go]: Oh, thanks for this comment. Makes sense. Fixed :)

**Comment by bryk** [comment]
Thanks for the review :)

**Pull request merged**